xdsl/printer.py

```diff
@@ -641,13 +641,17 @@ def _print_attr_string(self, attr_tuple: tuple[str, Attribute]) -> None:
             self.print(f'"{attr_tuple[0]}" = ')
             self.print_attribute(attr_tuple[1])
 
-    def print_op_attributes(self, attributes: dict[str, Attribute]) -> None:
-        if len(attributes) == 0:
-            return
+    def print_op_attributes(
+        self, attributes: dict[str, Attribute], reserved_attr_names: Iterable[str] = []
+    ) -> None:
+        attribute_list = list(
+            i for i in attributes.items() if i[0] not in reserved_attr_names
+        )
```

Comment:
Is this not just a [filter](https://www.programiz.com/python-programming/methods/built-in/filter).

Reply:
That's the "pythonic" way of writing a filter

Reply:
I was thinking of `list(filter(lambda i: i[0] not in reserved_attr_names, attributes))` or sth similar, but that's just a minor comment.

Reply:
`filter` is significantly slower than list comprehensions, unfortunately

Reply:
I wouldn't argue that we should use `filter` because of the speed (unless in the hot section of the code), but rather because of style though.

Reply:
They're somewhat related but sure